app/controllers/staff_dashboard_controller.rb

```diff
@@ -1,15 +1,7 @@
-class StaffDashboardController < ApplicationController
-  before_action :current_user, :ensure_admin
+class StaffDashboardController < StaffAreaController
 
   def index
-  end
-
-  private
-
-  def ensure_admin
-    unless admin?
-      redirect_to '/' and return
-    end
+    redirect_to staff_dashboard_index_path
```

Comment:
Is this an infinite redirect loop? Isn't this index action `staff_dashboard_index_path`